lib/image_optim/worker/jpegrecompress.rb

```diff
@@ -17,8 +17,19 @@ def self.init(image_optim, options = {})
       end.join(', ')
 
       QUALITY_OPTION =
-      option(:quality, 3, "JPEG quality preset: #{quality_names_desc}") do |v|
-        OptionHelpers.limit_with_range(v.to_i, 0...QUALITY_NAMES.length)
+        option(:quality, 3, "JPEG quality preset: #{quality_names_desc}") do |v|
+          OptionHelpers.limit_with_range(v.to_i, 0...QUALITY_NAMES.length)
+        end
+
+      METHOD_OPTION = option(:method, 'ssim', 'Compression method: '\
+          '`mpe` - Mean pixel error, '\
+          '`ssim` - Structural similarity, '\
+          '`ms-ssim` - Multi-scale structural similarity (slow!), '\
+          '`smallfry` - Linear-weighted BBCQ-like (may be patented)') do |v|
+        unless %w[mpe ssim ms-ssim smallfry].include? v
+          warn "Unknown method for jpegrecompress: #{v}"
```

Comment:
I think it is better to return the default value, when it is invalid. Can be done as for `Jpegoptim::MAX_QUALITY_OPTION`.